src/auth/auth.service.ts

```diff
@@ -64,6 +64,14 @@ export class AuthService {
    * @param user User object which gets from database
    */
   async register(user: RegisterUserDTO): Promise<Message> {
+    let check = await this.usersService.getUsersByQuery(user.code);
```

Comment:
Security leak - only return that "mail/code" already in use. Otherwise as a attacker I can generate a table with mails and codes in use.